src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -175,7 +176,8 @@
               "JDBC_TREAT_DECIMAL_AS_INT",
               "JDBC_ENABLE_COMBINED_DESCRIBE",
               CLIENT_ENABLE_CONSERVATIVE_MEMORY_USAGE,
-              CLIENT_VALIDATE_DEFAULT_PARAMETERS));
+              CLIENT_VALIDATE_DEFAULT_PARAMETERS,
+              "SNOWPARK_LAZY_ANALYSIS"));
```

Comment:
Why do we have to add this here ? In your previous PR, you added the logic to store all the other params in a Map and added a getter for it, right ?

Reply:
Good question, please look here: https://github.com/snowflakedb/snowflake-jdbc/blob/1cc5a8839267f8fc0691b58551bb9e3d08a4e5dc/src/main/java/net/snowflake/client/core/SessionUtil.java#L1261 Parameters that are not listed in these sets are directly ignored.
I tried to add non-listed parameters into the Map<String, Object> as well, but found another issue. Some non-listed parameter have corresponding handler in updateSfDriverParamValues, for example here: https://github.com/snowflakedb/snowflake-jdbc/blob/1cc5a8839267f8fc0691b58551bb9e3d08a4e5dc/src/main/java/net/snowflake/client/core/SessionUtil.java#L1359 
I'm not sure why this parameter is not listed as int/string/boolean parameter, but have a corresponding handler in the updateSfDriverParamValues function. So I just added Snowpark parameter to the list, which is simpler.